### PR TITLE
Coverage CI created, hopefully it works first time lol

### PR DESCRIPTION
**Pull request by oddbookworm**
I wanted to see what would happen if we generate coverage in our CI, so let's see what happens lol

### CODE REVIEW

**Comment by ankith26** [comment]
This PR is now passing with the meson buildconfig, I rebased the branch to get the fix from #2996 

The PR looks ready to me, though @oddbookworm if you want to experiment more, feel free to. I looked for actions that can integrate with `lcov` and put things out as a github comment, and I found https://github.com/marketplace/actions/code-coverage-report

**Comment by oddbookworm** [comment]
That looks super interesting. I’ll look into using that when I get home

File: .github/workflows/build-ubuntu-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        sudo apt-get install lcov -y
+        sudo apt-get install libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev libfreetype6-dev libportmidi-dev python3-dev -y
+        pip3 install --upgrade pip
+        pip3 install meson-python ninja cython "sphinx<=7.2.6"  # because we are doing --no-build-isolation
+        pip3 install numpy>=1.21.0
+
+    - name: Build with coverage hooks and install
+      id: build
+      run: |
+        pip3 install -e ${{ github.workspace }} --no-build-isolation -Cbuild-dir=${{ github.workspace }}/.mesonpy-rel -Csetup-args=-Dcoverage=true
```
Comment (ankith26) [inline .github/workflows/build-ubuntu-coverage.yml:75]: Any specific reason you made it more explicit with `${{ github.workspace }}` usage?

**Review by ankith26 — APPROVED** [review]
Left a review and a question but apart from that this PR LGTM, thanks! :moon_cake:

File: .github/workflows/build-ubuntu-coverage.yml (hunk 1 above)
Comment (oddbookworm) [reply .github/workflows/build-ubuntu-coverage.yml:75]: That was a leftover from my adventures with the other things I was trying. I was thinking maybe something was in a weird spot somewhere, so I made sure I knew absolutely where everything was

File: .github/workflows/build-ubuntu-coverage.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # gcov/lcov only gets C coverage
+      - 'src_py/**'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-ubuntu-coverage.yml'
+
+  # the github release drafter can call this workflow
+  workflow_call:
```
[the remark this answers is not in the document]
Comment (oddbookworm) [reply .github/workflows/build-ubuntu-coverage.yml:42]: oops

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :+1:

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

**Pull request merged**